### DISCUSSION TITLE
convert signed 16 bit image to unsigned

### DISCUSSION THREAD

**Discussion by manylegged** [ImageMagick/ImageMagick#1930] [General] [answered]
I would like to convert a signed 16 bit greyscale image (containing elevation data) into an unsigned format. There is an old thread about this here: https://imagemagick.org/discourse-server/viewtopic.php?t=27262 but unfortunately the OP never posted their solution.

I tried using the -fx option and this kind of works, but there is a jagged edge around the border between the negative and positive areas, as if I'm off by a few bits.
```convert -fx '(p < 0.5) ? (p + 0.5) : (p - 0.5)' signed_16bit.tif unsigned_16bit.png```

Any suggestions?

I'm using imagemagick through cygwin if it matters

**Comment by snibgo** [comment, marked as answer]
What version of IM?

Your syntax is wrong. You should read the input, then "-fx", then write the output.

Assuming your IM is Q16, "-evaluate add 50%" should do it. Or it might be "-evaluate subtract 50%". If using Windows BAT scripts, double the percent: %%.

But the proper way is to use "-define quantum:format=signed" or unsigned. See https://www.imagemagick.org/script/command-line-options.php#define.

**Reply by manylegged** [reply]
I was able to get the right output with the AddModulus function:

```convert signed_16bit.tif -evaluate AddModulus '50%'  unsigned.png```

This is with Version: ImageMagick 6.9.10-11 Q16 x86_64 2019-12-10 https://www.imagemagick.org
Thanks!

**Reply by snibgo** [reply]
Yes, AddModulus, thanks for the correction. My brain isn't working today.
